src/components/structures/MatrixChat.js

```diff
@@ -584,6 +588,44 @@ module.exports = React.createClass({
     _onLoadCompleted: function() {
         this.props.onLoadCompleted();
         this.setState({loading: false});
+
+        // set up the right theme.
+        // XXX: this will temporarily flicker the wrong CSS.
+        dis.dispatch({
+            action: 'set_theme',
+            value: UserSettingsStore.getSyncedSetting('theme')
+        });
+    },
+
+    /**
+     * Called whenever someone changes the theme
+     */
+    _onSetTheme: function(theme) {
```

Comment:
Not sure this belongs here. MatrixChat is too much of a dumping-ground as it is. Could it go somewhere in riot-web?